[ai-gm] GPT-5 models cause "Unsupported parameter: 'max_tokens' is not supported"

Selecting a GPT-5 model in **Settings** and then sending a chat message results in the following lines in the browser console and no response from the AI GM is received:

```console
api.openai.com/v1/chat/completions:1  Failed to load resource: the server responded with a status of 400 ()
installHook.js:1 Error getting GM response: BadRequestError: 400 Unsupported parameter: 'max_tokens' is not supported with this model. Use 'max_completion_tokens' instead.
    at _APIError.generate (chunk-JMWJHOW7.js?v=b6725208:1422:14)
    at OpenAI.makeStatusError (chunk-JMWJHOW7.js?v=b6725208:874:21)
    at OpenAI.makeRequest (chunk-JMWJHOW7.js?v=b6725208:919:24)
    at async getGMResponse (orchestration.ts:195:22)
    at async handleSubmit (ChatPanel.tsx:70:24)
overrideMethod @ installHook.js:1
```

The GPT-4o models are unaffected by this issue because they support the `max_tokens` parameter.

--- a/ai-gm/src/app/components/SettingsDrawer.tsx
+++ b/ai-gm/src/app/components/SettingsDrawer.tsx
@@ -4,6 +4,9 @@ import { formatLabel } from '@/lib/ui/formatting'
 export default function SettingsDrawer() {
   const { isSettingsOpen, setSettingsOpen, settings, updateSettings } = useAppStore()
 
+  // GPT-5 models don't support temperature or max_tokens parameters
+  const isGPT5Model = settings.model.toLowerCase().includes('gpt-5')
+
   if (!isSettingsOpen) return null
 
   return (
@@ -76,10 +79,13 @@ export default function SettingsDrawer() {
                     const value = parseFloat(e.target.value)
                     updateSettings({ temperature: isNaN(value) ? 1 : value })
                   }}
-                  className="input w-full"
+                  disabled={isGPT5Model}
+                  className="input w-full disabled:cursor-not-allowed disabled:opacity-50"
                 />
                 <p className="mt-1 text-xs text-text-muted">
-                  Higher values make output more random, lower values more focused
+                  {isGPT5Model
+                    ? 'Not supported by GPT-5 models'
+                    : 'Higher values make output more random, lower values more focused'}
                 </p>
               </div>
               <div>
@@ -98,10 +104,13 @@ export default function SettingsDrawer() {
                     })
                   }
                   placeholder="Default (model-specific)"
-                  className="input w-full"
+                  disabled={isGPT5Model}
+                  className="input w-full disabled:cursor-not-allowed disabled:opacity-50"
                 />
                 <p className="mt-1 text-xs text-text-muted">
-                  Limits the length of GM responses
+                  {isGPT5Model
+                    ? 'Not supported by GPT-5 models (uses max_completion_tokens instead)'
+                    : 'Limits the length of GM responses'}
                 </p>
               </div>
             </div>

--- a/ai-gm/src/lib/openai/orchestration.ts
+++ b/ai-gm/src/lib/openai/orchestration.ts
@@ -192,7 +192,8 @@ export async function getGMResponse(request: GMRequest): Promise<GMResponse> {
     iterations++
 
     // Determine which token parameter to use based on model
-    // GPT-5 models require max_completion_tokens, GPT-4 models use max_tokens
+    // GPT-5 models require max_completion_tokens and don't support temperature
+    // GPT-4 models use max_tokens and support temperature
     const isGPT5Model = model.toLowerCase().includes('gpt-5')
     const tokenParam = isGPT5Model ? 'max_completion_tokens' : 'max_tokens'
 
@@ -202,7 +203,8 @@ export async function getGMResponse(request: GMRequest): Promise<GMResponse> {
       messages: currentMessages,
       tools: tools as OpenAI.Chat.Completions.ChatCompletionTool[],
       parallel_tool_calls: false, // Force sequential tool calls for determinism
-      ...(settings?.temperature !== undefined && { temperature: settings.temperature }),
+      // GPT-5 models don't support temperature parameter
+      ...(!isGPT5Model && settings?.temperature !== undefined && { temperature: settings.temperature }),
       ...(settings?.max_tokens !== undefined && { [tokenParam]: settings.max_tokens }),
     })
 

